system/timers.c

```diff
@@ -85,6 +91,33 @@ static void correct_tsc(void)
        clks_per_msec = run_time / 50;
     }
 }
+#elif defined(__loongarch_lp64)
+static void correct_tsc(void)
+{
+    uint64_t start, end, excepted_ticks, current_ticks, calc_base_freq, clock_multiplier, clock_divide;
+    uint64_t num_millisec = 50, millisec_div = 1000;
+
+    //
+    // Get stable count frequency
+    //
+    calc_base_freq   = __cpucfg(0x4);
+    clock_multiplier = __cpucfg(0x5) & 0xFFFF;
+    clock_divide     = (__cpucfg(0x5) >> 16) & 0xFFFF;
+
+    excepted_ticks = (((calc_base_freq * clock_multiplier) / clock_divide) * num_millisec) / millisec_div;
+
+    __asm__ __volatile__("rdtime.d %0, $zero":"=r"(current_ticks):);
```

Comment:
The `__rdtime_d` intrinsic could be more readable. But I think maybe it's that you don't want to read a node ID along with the tick counter?

Reply:
I just don't want to add more code, `__rdtime_d` returned not a value but a structure, so I think it is easy to use `__rdtime_d`. BTW, this instruction is reads out the Stable Counter, not the node counter, Stable Counter is a global stable timer, anyone using this instruction can only read the same stable counter.